### PR TITLE
feat(ui): enforce valid git branch names in worktree dialogs

### PR DESCRIPTION
**Pull request by hayke102**
## Summary
- Worktree dialogs (`w` key and the native-git create-workspace sub-dialog) now guide the user toward a valid git branch name as they type: spaces become `-`, and chars git forbids anywhere in a ref (`~ ^ : ? * [ \` and ASCII control chars) are dropped live. `/` stays legal.
- Rules that can't be fixed silently (empty, leading `-/./`, trailing `./lock/`, `..`, `@{`, `//`, lone `@`) are checked on submit and surface as a friendly inline error in the dialog instead of a cryptic `git worktree add` failure.
- Two new helpers in `internal/workspace/provider.go`: `SanitizeBranchInput` (live-transform) and `ValidateBranchName` (submit-check), with full test coverage.

## Test plan
- [x] `make build`
- [x] `go test -race ./internal/ui/... ./internal/workspace/...`
- [ ] Manual TUI smoke test — press `w` on a worktree-enabled repo and try:
  - `my cool branch` → shows `my-cool-branch` live
  - `feat~evil` → tilde dropped as typed
  - `feature/login` → slash preserved; path preview shows `feature-login` dir
  - `-bad` + Enter → inline red error, no dispatch
  - `a..b` + Enter → inline red error
  - valid name + Enter → worktree created as before
- [ ] Repeat the same typing checks in the native-git create-workspace sub-dialog

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Live branch-name sanitization while typing (spaces -> dashes, disallowed chars removed) and improved inline validation to prevent cryptic git errors on submit.
* **Tests**
  * Added tests covering input sanitization, cursor behavior during edits, and branch-name validation for various edge cases.

### CODE REVIEW

File: internal/workspace/provider.go [hunk 1]
```diff
@@ -325,6 +325,68 @@ func SanitizeBranchName(branch string) string {
...
+		case r < 0x20 || r == 0x7f:
+			// drop ASCII control chars
+		case r == '~' || r == '^' || r == ':' || r == '?' || r == '*' || r == '[' || r == '\\':
+			// drop chars git forbids anywhere in a ref
+		default:
+			b.WriteRune(r)
+		}
+	}
+	return b.String()
+}
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/workspace/provider.go:371]: Fixed — added backtick to the drop set in SanitizeBranchInput (cee21f8).

File: internal/workspace/provider.go [hunk 2]
```diff
@@ -325,6 +325,68 @@ func SanitizeBranchName(branch string) string {
...
+		return "Branch name cannot be '@'"
+	}
+	switch branch[0] {
+	case '-':
+		return "Branch name cannot start with '-'"
+	case '/':
+		return "Branch name cannot start with '/'"
+	case '.':
+		return "Branch name cannot start with '.'"
+	}
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/workspace/provider.go:388]: Fixed — ValidateBranchName now validates each '/'-separated component for leading '.' (cee21f8).

File: internal/workspace/provider.go [hunk 3]
```diff
@@ -325,6 +325,68 @@ func SanitizeBranchName(branch string) string {
...
+	switch branch[len(branch)-1] {
+	case '.':
+		return "Branch name cannot end with '.'"
+	case '/':
+		return "Branch name cannot end with '/'"
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/workspace/provider.go]: Fixed — per-component trailing '.' and '.lock' are now rejected alongside the leading-'.' check (cee21f8).

File: internal/workspace/provider_test.go [hunk 4]
```diff
@@ -68,6 +68,86 @@ func TestParseWorktreePorcelain(t *testing.T) {
...
+	invalid := []struct {
+		name  string
+		input string
+	}{
+		{"empty", ""},
+		{"at sign alone", "@"},
+		{"leading dash", "-bad"},
+		{"leading slash", "/bad"},
+		{"leading dot", ".bad"},
+		{"trailing dot", "bad."},
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/workspace/provider_test.go:152]: Fixed — added 'component leading dot', 'component trailing dot', and 'component .lock' to the invalid-cases table (cee21f8).

File: internal/ui/workspace_picker.go [hunk 5]
```diff
@@ -228,6 +228,10 @@ func (d *WorktreeDialog) Update(msg tea.Msg) (*WorktreeDialog, tea.Cmd) {
 		d.baseBranchInput, cmd = d.baseBranchInput.Update(msg)
 	case focusNewBranch:
 		d.newBranchInput, cmd = d.newBranchInput.Update(msg)
+		if sanitized := workspace.SanitizeBranchInput(d.newBranchInput.Value()); sanitized != d.newBranchInput.Value() {
+			d.newBranchInput.SetValue(sanitized)
+			d.newBranchInput.CursorEnd()
+		}
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/ui/workspace_picker.go]: Fixed — introduced SanitizeBranchInputWithCursor that returns the adjusted cursor by counting dropped runes before the old position; the dialog now uses SetCursor(newPos) instead of CursorEnd (cee21f8).

File: internal/ui/workspace_create.go [hunk 6]
```diff
@@ -213,12 +213,21 @@ func (d *CreateWorkspaceDialog) Update(msg tea.Msg) (*CreateWorkspaceDialog, tea
...
 	var cmd tea.Cmd
+	branchTouched := false
 	if d.isNative {
 		d.branchInput, cmd = d.branchInput.Update(msg)
+		branchTouched = true
 	} else if d.focusIndex == 0 {
 		d.nameInput, cmd = d.nameInput.Update(msg)
 	} else {
 		d.branchInput, cmd = d.branchInput.Update(msg)
+		branchTouched = true
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/ui/workspace_create.go:224]: Kept as-is. The characters we strip (space, `~ ^ : ? * [ \ \` `, ASCII controls) are unambiguously invalid in git refs, and custom-shell providers in practice still template {{branch}} into a git command. Leaving the custom-shell path symmetric with the native path avoids two divergent behaviors for essentially the same input. Happy to revisit if a concrete provider breaks on it.

File: internal/ui/workspace_create.go [hunk 7]
```diff
@@ -213,12 +213,21 @@ func (d *CreateWorkspaceDialog) Update(msg tea.Msg) (*CreateWorkspaceDialog, tea
...
+	}
+	if branchTouched {
+		if sanitized := workspace.SanitizeBranchInput(d.branchInput.Value()); sanitized != d.branchInput.Value() {
+			d.branchInput.SetValue(sanitized)
+			d.branchInput.CursorEnd()
+		}
```
[the remark this answers is not in the document]
Comment (hayke102) [reply internal/ui/workspace_create.go]: Fixed — same SanitizeBranchInputWithCursor helper used here; cursor is preserved across drops (cee21f8).

**Review by denniszag — APPROVED** [review]
